test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -1731,6 +1731,20 @@ public static boolean indexExists(String index, Client client) {
         return getIndexResponse.getIndices().length > 0;
     }
 
+    public static void ensureIndexExists(String index) {
+        ensureIndexExists(index, SAFE_AWAIT_TIMEOUT.seconds(), SAFE_AWAIT_TIMEOUT.timeUnit());
+    }
+
+    public static void ensureIndexExists(String index, long timeout, TimeUnit unit) {
+        safeGet(
+            clusterAdmin().prepareHealth(new TimeValue(timeout, unit), index)
+                .setIndicesOptions(IndicesOptions.LENIENT_EXPAND_OPEN_CLOSED)
+                .execute(),
+            timeout,
+            unit
+        );
+    }
```

Comment:
Raised this PR as a draft since I am not sure whether this is what we want. There are scenarios that are not covered by this, e.g. assertBusy for index _not_ exists, assertBusy for an index exists on a remote cluster. 

For the former, I wonder whether it is easier to change the existing `indexExists` method to always talk to the master node. For the later, I don't have a great suggestion.

Reply:
In an integ test involving multiple clusters we have a client for each cluster so we should be able to do the same health API call there.

For the assertBusy/assertFalse waits that I could find it looks like it's ok to wait on any node - we're just checking that the index deletion has been committed, not that it has been applied everywhere, so need no change. For instance if the next thing the test does involves another cluster state update then that will of course wait for the previous one to complete. If that weren't the case, a `GET _cluster/health?wait_for_events=LANGUID` would do the trick I think.

Reply:
Thanks for the suggestions. Yeah the assertFalse case does not seem to be problematic. Unlike assertTrue case, the tests do not anything more against the deleted index. For the remote cluster case, I made the variant of the method to take a client parameter.